src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/TestAzureContainerRegistryWebhook.cs

```diff
@@ -0,0 +1,72 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsDiagnostic.Test, ContainerRegistryWebhookNoun, DefaultParameterSetName = ResourceIdParameterSet)]
+    [OutputType(typeof(PSContainerRegistryEventInfo))]
+    public class TestAzureContainerRegistryWebhook : ContainerRegistryCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Webhook Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(WebhookNameAlias, ResourceNameAlias)]
+        public string Name { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string RegistryName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = WebhookObjectParameterSet, ValueFromPipeline = true, HelpMessage = "Container Registry Object.")]
+        [ValidateNotNull]
+        public PSContainerRegistryWebhook Webhook { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The container registry Webhook resource id")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ResourceIdAlias)]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (string.Equals(ParameterSetName, WebhookObjectParameterSet))
+            {
+                ResourceId = Webhook.Id;
+            }
+            if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
+            {
+                string resourceGroup, registryName, childResourceName;
+                if(!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName)
+                    || string.IsNullOrEmpty(childResourceName))
+                {
+                    WriteInvalidResourceIdError(InvalidWebhookResourceIdErrorMessage);
+                    return;
+                }
+
+                ResourceGroupName = resourceGroup;
+                Name = childResourceName;
+                RegistryName = registryName;
+            }
+
+            var eventInfo = RegistryClient.PingWebhook(ResourceGroupName, RegistryName, Name);
+            WriteObject(new PSContainerRegistryEventInfo(eventInfo));
```

Comment:
@Jinhuafei usually `Test-*` cmdlets return a `bool` representing whether or not the resource of interest exists

Reply:
This is to trigger a ping webhook event, so returning an event object makes more sense.

Reply:
👍